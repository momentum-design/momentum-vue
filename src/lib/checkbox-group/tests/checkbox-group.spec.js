import { mount } from '@vue/test-utils';
import CheckboxGroup from '../index.vue';
import Checkbox from '../../checkbox/index.vue';
import Label from '../../label/index.vue';
import Vue from 'vue';

Vue.component(Checkbox.name, Checkbox);
Vue.component(Label.name, Label);

describe('CheckboxGroup', () => {
  it('should match snapshot', () => {
    const wrapper = mount(CheckboxGroup, {
      slots: {
        default: `<md-checkbox htmlId="test" label="test" />`
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should handle default true Values', () => {
    const wrapper = mount(CheckboxGroup, {
      propsData: {
        values: ['me']
      },
      slots: {
        default: `<md-checkbox htmlId="testCheckbox1" label="me" value="me" />
<md-checkbox htmlId="testCheckbox2" label="you" value="you" />
<md-checkbox htmlId="testCheckbox3" label="us" value="us" />
`
      }
    });

    expect(wrapper.findAll(Checkbox).at(0).props().checked).toEqual(true);
    expect(wrapper.findAll(Checkbox).at(1).props().checked).toEqual(false);
  });

  it('should handle Change Events', () => {
    const wrapper = mount(CheckboxGroup, {
      slots: {
        default: `<md-checkbox htmlId="testCheckbox1" label="me" value="me" />
<md-checkbox htmlId="testCheckbox2" label="you" value="you" />
<md-checkbox htmlId="testCheckbox3" label="us" value="us" />
`
      }
    });

    wrapper
      .find('#testCheckbox1')
      .trigger('change');

    expect(wrapper.findAll(Checkbox).at(0).props().checked).toEqual(true);
    expect(wrapper.findAll(Checkbox).at(1).props().checked).toEqual(false);
  });
});

