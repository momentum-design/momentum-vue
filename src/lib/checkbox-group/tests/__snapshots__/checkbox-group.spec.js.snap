// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup should match snapshot 1`] = `
<div class="md-checkbox-group">
  <div class="md-input-group md-checkbox"><input type="checkbox" id="test" name="" tabindex="0" class="md-input md-checkbox__input" value=""> <label for="test" class="md-checkbox__label"><span>test</span></label> </div>
</div>
`;
